Add support for GRPC API to H2O-3

st-pasha opened the pull request:
Related issues: [PUBDEV-2370](https://0xdata.atlassian.net/browse/PUBDEV-2370), [PUBDEV-3650](https://0xdata.atlassian.net/browse/PUBDEV-3650)

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

st-pasha commented:
@h2o-ops please test!

mmalohlava commented:
The project import works in my idea. 

I realized one issue: putting grpc module into assembly is going to increase `h2o.jar` size (bringing problems with PyPi).

navdeep-G commented:
@mmalohlava @st-pasha Tiny note: We just get the jar size small enough for PyPi. Currently, we are ~3MB under the limit for PyPi.

st-pasha commented:
We don't have to put it into the "general" assembly, just into the one that is going to be shipped with `py2o`. And for that assembly we can exclude many other features such as hadoop / parquet / orc / flow / rest / etc.

mmalohlava commented:
Ok, sounds good for me - let's remove it from main `h2o.jar` assembly for now.

mmalohlava commented on settings.gradle:
```diff
@@ -6,7 +6,8 @@ include 'h2o-web'
 include 'h2o-r'
 include 'h2o-py'
-include 'h2o-assembly'
+include 'h2o-assemblies:main'
```
interesting - does it work?

st-pasha replied:
as far as I can see, it does. The gradle build completes successfully; builds both versions of h2o.jar; and both jars can start without throwing any exceptions

st-pasha commented:
Removed GRPC from "main" h2o assembly; added it to the "py2o" assembly. Also for the latter excluded hdfs / parquet / flow / avro support -- it reduces the size of the jar file down to 18Mb only.

mmalohlava commented on h2o-app/build.gradle:
```diff
@@ -4,12 +4,10 @@
 description = "H2O Application Runner"
 
 dependencies {
-  compile project(":h2o-web")
```
This is breaking change!! :-(
